dragonfly/item/item.go

```diff
@@ -79,6 +85,8 @@ type UseContext struct {
 	// NewItem is the item that is added after the item is used. If the player no longer has an item in the
 	// hand, it'll be added there.
 	NewItem Stack
+	// SurvivalDedicatedItem will add any new items only in survival mode.
+	SurvivalDedicatedItem bool
```

Comment:
I feel like the name of this field is rather ambiguous. If I hadn't read the rest of the code, I wouldn't know what this did. Might be good to rename it to something more clear, even if the field name gets longer.